Formula/talisman.rb

```diff
@@ -0,0 +1,20 @@
+class Talisman < Formula
+  desc "Tool to detect and prevent secrets from getting checked in"
+  homepage "https://thoughtworks.github.io/talisman/"
+  url "https://github.com/thoughtworks/talisman",
+      :using    => :git,
+      :tag      => "v1.0.0",
+      :revision => "dd7b5898f7c4d9013d23d4dbd34685611524ac20"
+
+  depends_on "go" => :build
+  depends_on "gox" => :build
+
+  def install
+    system "./build"
+    bin.install "./talisman_darwin_amd64" => "talisman"
+  end
+
+  test do
+    system bin/"talisman", "--version"
```

Comment:
We need a test that exercises the some of the functionality of the app. Version checks or usage checks (foo --version or foo --help) are not sufficient, as explained in the formula cookbook.

In most cases, a good test would involve running a simple test case: run #{bin}/foo input.txt.

- Then you can check that the output is as expected (with assert_equal or assert_match on the output of shell_output)
- You can also check that an output file was created, if that is expected: assert_predicate testpath/"output.txt", :exist?

Some advice for specific cases:

- If the formula is a library, compile and run some simple code that links against it. It could be taken from upstream's documentation / source examples.
- If the formula is for a GUI program, try to find some function that runs as command-line only, like a format conversion, reading or displaying a config file, etc.
- If the software cannot function without credentials, a test could be to try to connect with invalid credentials (or without credentials) and confirm that it fails as expected.
- Same if the software requires a virtual machine, docker instance, etc. to be running.

Reply:
Add test which will test the actual functionality of the binary itself